README.md

```diff
@@ -1,14 +1,192 @@
+
+#### The `tls-certificates` interface
+
+The `tls-certificates` interface could be used with any charm that provides TLS certificate lifecycle management functionality, following [`this specification`](https://github.com/canonical/charm-relation-interfaces/tree/main/docs/json_schemas/tls_certificates/v1). One example is the [`self-signed-certificates`](https://github.com/canonical/self-signed-certificates-operator) operator by Canonical.
+
+Note that TLS can be enabled in three different modes:
+
+- For Kafka Connect REST interface only
+- For the relation between Apache Kafka cluster and Kafka Connect
+- For both the REST interface and the relation
+
+To enable TLS on the Kafka Connect REST interface, first deploy the TLS charm and relate it to the Charmed Kafka Connect application:
+
+```bash
+juju deploy self-signed-certificates --channel=stable
+juju integrate self-signed-certificates kafka-connect
+```
+
+To enable TLS on the relation between the Apache Kafka and Kafka Connect clusters: 
+
+```bash
+juju integrate self-signed-certificates kafka
+```
+
+To disable TLS on each interface, remove their respective relations:
+
+```bash
+juju remove-relation kafka-connect self-signed-certificates
+juju remove-relation kafka self-signed-certificates
+```
+
+> Note: The TLS settings here are for self-signed-certificates which are not recommended for production clusters, the `tls-certificates-operator` charm offers a variety of configurations, read more on the TLS charm [here](https://charmhub.io/tls-certificates-operator)
+
+## Monitoring
+
+The Charmed Kafka Connect Operator comes with the [JMX exporter](https://github.com/prometheus/jmx_exporter/).
+The metrics can be queried by accessing the `http://<unit-ip>:9100/metrics` endpoints.
+
+Additionally, the charm provides integration with the [Canonical Observability Stack](https://charmhub.io/topics/canonical-observability-stack).
+
+Deploy the `cos-lite` bundle in a Kubernetes environment. This can be done by following the
+[deployment tutorial](https://charmhub.io/topics/canonical-observability-stack/tutorials/install-microk8s).
+Since the Charmed Kafka Connect Operator is deployed on a machine environment, it is needed to offer the endpoints
+of the COS relations. The [offers-overlay](https://github.com/canonical/cos-lite-bundle/blob/main/overlays/offers-overlay.yaml)
+can be used, and this step is shown in the COS tutorial.
+
+Next, deploy [grafana-agent](https://charmhub.io/grafana-agent) and follow the
+[tutorial](https://discourse.charmhub.io/t/using-the-grafana-agent-machine-charm/8896)
+to relate it to the COS Lite offers.
+
+Now, relate `kafka-connect` with the `grafana-agent`:
```

Comment:
Suggestion: Our general guidance is to use `integrate` for the verb. We still can use `relation` for nouns.

```suggestion
Now, integrate `kafka-connect` and `grafana-agent` charms:
```